image/gimp_brush.ksy

```diff
@@ -0,0 +1,43 @@
+meta:
+  id: gimp_brush
+  title: GIMP (GNU Image Manipulation Program) brush version 2 file
+  file-extension: gbr
+  license: CC0-1.0
+  endian: be
+doc-ref: https://gitlab.gnome.org/GNOME/gimp/-/raw/4416313/devel-docs/gbr.txt
+seq:
+  - id: len_header
+    type: u4
+  - id: header
+    type: header
+    size: len_header - len_header._sizeof
+types:
+  header:
+    seq:
+      - id: version
+        type: u4
+      - id: width
+        type: u4
+      - id: height
+        type: u4
```

Comment:
There are assertions for these values in the GIMP code ([`gimpbrush-load.c:191-192`](https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-load.c#L191-L192)):

```c
  if (header.width  > GIMP_BRUSH_MAX_SIZE ||
      header.height > GIMP_BRUSH_MAX_SIZE ||
      /* ... */)
    {
      g_set_error (error, GIMP_DATA_ERROR, GIMP_DATA_ERROR_READ,
                   _("Fatal parse error in brush file: %dx%d over max size."),
                   header.width, header.height);
      return NULL;
    }
```

[`gimpbrush-header.h:24`](https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-header.h#L24):

```c
#define GIMP_BRUSH_MAX_SIZE 10000 /* Max size in either dimension in px */
```

[`gimpbrush-load.c:170-182`](https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-load.c#L170-L182):

```c
  if (header.width == 0)
    {
      g_set_error (error, GIMP_DATA_ERROR, GIMP_DATA_ERROR_READ,
                   _("Fatal parse error in brush file: Width = 0."));
      return NULL;
    }


  if (header.height == 0)
    {
      g_set_error (error, GIMP_DATA_ERROR, GIMP_DATA_ERROR_READ,
                   _("Fatal parse error in brush file: Height = 0."));
      return NULL;
    }
```

If we want to ensure that the GIMP brush is compatible with GIMP, I'd add the assertions too:

```suggestion
      - id: width
        type: u4
        valid:
          min: 1
          max: 10000
        doc-ref:
          - https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-load.c#L170 # valid/min
          - https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-header.h#L24 # valid/max
      - id: height
        type: u4
        valid:
          min: 1
          max: 10000
        doc-ref:
          - https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-load.c#L177 # valid/min
          - https://github.com/GNOME/gimp/blob/9b6d59f/app/core/gimpbrush-header.h#L24 # valid/max
```